### PR TITLE
[In progress] Git support

### PR DESCRIPTION
**Pull request by jeremypw**
Initial attempt to enable git awareness in Files.
The code is largely copied from the equivalent Code code.

### CODE REVIEW

File: libcore/gof-directory-async.vala [hunk 1]
```diff
@@ -235,6 +249,30 @@ public class Async : Object {
...
 
+    public void update_git_status () {
+        if (git_repo == null) {
+            return;
+        }
+
+        var options = new Ggit.StatusOptions (Ggit.StatusOption.INCLUDE_UNTRACKED, Ggit.StatusShow.INDEX_AND_WORKDIR, null);
+
+        try {
+            /* FIXME This seg faults for unknown reasons */
```
Comment (vjr) [inline libcore/gof-directory-async.vala]: Have you tried changing the `check_each_git_status` function from private to public?

Comment (jeremypw) [reply libcore/gof-directory-async.vala]: Thanks Vishal - David Hewitt advised that the problem is that I had not initialised the Ggit library before using this function.  Confusingly, all the other Ggit library functions that I used worked OK so this was not easy to spot.

**Comment by jeremypw** [comment]
I presume this is failing Travis CI because need to add libgit2-glib-1.0 dependency to "deb-packaging" branch.

**Comment by jeremypw** [comment]
Withdrawing in favour of a plugin approach.

**Pull request closed without merging**